integration/nwo/network.go

```diff
@@ -1607,3 +1657,34 @@ func (n *Network) GenerateCoreConfig(p *Peer) {
 	err = t.Execute(io.MultiWriter(core, pw), n)
 	Expect(err).NotTo(HaveOccurred())
 }
+
+// ChaincodeConnectionDir returns the path to the configuration directory for the specified
+// chaincode service.
+func (n *Network) ChaincodeConnectionDir(chaincode *Chaincode) string {
+	return filepath.Join(n.RootDir, "chaincode-connections", chaincode.ID())
+}
+
+// ChaincodeConnectionPath returns the path to the chaincode connction.json document
+// for the specified chaincode service
+func (n *Network) ChaincodeConnectionPath(chaincode *Chaincode) string {
+	return filepath.Join(n.ChaincodeConnectionDir(chaincode), "connection.json")
+}
+
+func (n *Network) GenerateChaincodeConnectionConfig(chaincode *Chaincode) {
+	err := os.MkdirAll(n.ChaincodeConnectionDir(chaincode), 0755)
+	Expect(err).NotTo(HaveOccurred())
+
+	core, err := os.Create(n.ChaincodeConnectionPath(chaincode))
+	Expect(err).NotTo(HaveOccurred())
+	defer core.Close()
+
+	t, err := template.New("peer").Funcs(template.FuncMap{
+		"Chaincode": func() *Chaincode { return chaincode },
+	}).Parse(n.Templates.ChaincodeConnectionTemplate())
+	Expect(err).NotTo(HaveOccurred())
+
+	time.Sleep(15 * time.Second)
+	pw := gexec.NewPrefixedWriter(fmt.Sprintf("[%s#core.yaml] ", chaincode.ID()), ginkgo.GinkgoWriter)
+	err = t.Execute(io.MultiWriter(core, pw), n)
+	Expect(err).NotTo(HaveOccurred())
+}
```

Comment:
These do not belong in the network. Your test should explicitly create the `connection.json` that you need.